cards3-template.js

```diff
@@ -1,19 +1,45 @@
 function makeCard(id) {
     // If id is invalid (out of range, etc)
+    //id must be a number
+    //id must be between 0 and 51
+
+    if(typeof id !== "number")
+    {
+        return null;
+    }
+
+    if(id < 0 || id > 51)
+    {
+        return null;
+    }
     // ...
```

Comment:
God job here! We would also want to check to make sure that we are returning an whole number, so we would want to add this to your makeCard function as well:

&& (num%1 === 0)

Other than that, this part looks great!